src/openrct2/ride/ride.c

```diff
@@ -8681,6 +8681,14 @@ void fix_invalid_vehicle_sprite_sizes()
             while (rideSpriteIndex != SPRITE_INDEX_NULL) {
                 rct_vehicle *vehicle = GET_VEHICLE(rideSpriteIndex);
                 rct_ride_entry_vehicle *vehicleEntry = vehicle_get_vehicle_entry(vehicle);
+
+                if (vehicle == NULL ||
+                    vehicleEntry == NULL ||
+                    vehicleEntry == (rct_ride_entry_vehicle*)-1)
+                {
+                    continue;
```

Comment:
Yes! Nice catch. Will fix later, feel free to do it yourself in the mean time if you want to get this merged.